snet_cli/identity.py

```diff
@@ -47,8 +47,8 @@ def __init__(self, w3, private_key):
                                                   curve=ecdsa.SECP256k1,
                                                   hashfunc=hashlib.sha256).get_verifying_key()
 
-        self.address = self.w3.toChecksumAddress(
-            "0x" + self.w3.sha3(hexstr=public_key.to_string().hex())[12:].hex())
+        self.address = self.w3.sha3(hexstr=public_key.to_string().hex())[12:].hex()
```

Comment:
Is this related to the nodejs client stub generation?

Reply:
Why have you removed toChecksumAddress?

Reply:
Hey,

I did not remove toChecksumAddress: if you check the very next line it says:
```
self.address = self.w3.toChecksumAddress(self.address) if self.address.startswith("0x") else self.w3.toChecksumAddress("0x" + self.address)
```
I'm just checking if the generated `self.address` already starts with `0x` in which case I'm not adding it twice. Reason is that the `.hex()` method is already returning addresses prefixed with `0x` on my system so I get this error in the constructor of Identity:
```
ValueError: when sending a str, it must be a hex string. Got: '0x0x63fe11d4634044c2949623c48cc695b5cb0d14de'
```

which means I get this error all the time. This only started happening recently and I haven't looked into it yet much but at the same time, the change is not disruptive: if the outputted address already starts with `0x`, don't add it again.